test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/TagHelperComponentTagHelperTest.cs

```diff
@@ -282,6 +304,43 @@ public void Init_LogsTagHelperComponentInitialized()
             Assert.Equal($"Tag helper component '{typeof(TestTagHelperComponent)}' processed.", sink.Writes[0].State.ToString(), StringComparer.Ordinal);
         }
 
+        [Fact]
+        public void Init_GetsTagHelperComponentPropertyActivator_FromRequestServices()
+        {
+            // Arrange            
+            var tagHelperContext = new TagHelperContext(
+                "head",
+                allAttributes: new TagHelperAttributeList(
+                    Enumerable.Empty<TagHelperAttribute>()),
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+
+            var output = new TagHelperOutput(
+                "head",
+                attributes: new TagHelperAttributeList(),
+                getChildContentAsync: (useCachedResult, encoder) => Task.FromResult<TagHelperContent>(
+                    new DefaultTagHelperContent()));
+
+            var testTagHelperComponentManager = new TagHelperComponentManager(new[]
+            {
+                new TestTagHelperComponent()
+            });
+
+            var testTagHelperComponentTagHelper = new TestTagHelperComponentTagHelper(
+                testTagHelperComponentManager,
+                NullLoggerFactory.Instance);
+
+            var viewContext = CreateViewContext(new DefaultHttpContext());
+            testTagHelperComponentTagHelper.ViewContext = viewContext;
+
+            // Act
+            testTagHelperComponentTagHelper.Init(tagHelperContext);
+
+            // Assert
+            Assert.NotNull(testTagHelperComponentTagHelper.PropertyActivator);
+            Assert.Same(viewContext.ViewData, testTagHelperComponentTagHelper.ViewContext.ViewData);
```

Comment:
No need to verify this piece. That logic is verified in a different test. Over-verifying can make test maintenance troublesome. Imagine we accidentally broke this behavior; instead of one test failing you'd have two, twice the work to fix with the same outcome.